src/common/components/Count/Count.tsx

```diff
@@ -42,37 +45,46 @@ const Count: FC<CountProps> = ({
   };
 
   const increaseCount = () => {
-    setCount((prev) => prev + 1);
+    const newCount = count + 1;
+    const newValidCount = getValidNumber(newCount.toString(), maxCount);
+    handleChangeAmount(newValidCount);
+    newCount > maxCount ? setError(true) : setError(false);
```

Comment:
Simplify

Reply:
Fixed - create single function for increment and decrement